tests/gateway.ts

```diff
@@ -4131,4 +4131,524 @@ describe("Gateway", () => {
+          .signers([nominatedAuthority])
+          .rpc();
+      }
+
+      const randomKeypair = anchor.web3.Keypair.generate();
+
+      // Fund the random keypair
+      await anchor.web3.sendAndConfirmTransaction(
+        conn,
+        new anchor.web3.Transaction().add(
+          anchor.web3.SystemProgram.transfer({
+            fromPubkey: wallet.publicKey,
+            toPubkey: randomKeypair.publicKey,
+            lamports: 10_000_000,
+          })
+        ),
+        [wallet]
+      );
+
+      try {
+        await gatewayProgram.methods
+          .acceptAuthority()
+          .accounts({
+            pda: pdaAccount,
+            newAuthority: randomKeypair.publicKey,
+          })
+          .signers([randomKeypair])
+          .rpc();
+        throw new Error("Expected error not thrown");
+      } catch (err) {
+        expect(err).to.be.instanceof(anchor.AnchorError);
+        expect(err.message).to.include("NoNominatedAuthority");
+      }
+    });
+  });
+
+  describe("Backward Compatibility After Migration", () => {
+    it("Should maintain all existing functionality after migration", async () => {
+      // First, ensure deposits are not paused
+      const pdaData = await gatewayProgram.account.pda.fetch(pdaAccount);
+      const currentAuthorityPubkey = pdaData.authority;
+
+      // Find the keypair for current authority
+      let currentAuthKeypair: anchor.web3.Keypair | null = null;
+      if (currentAuthorityPubkey.equals(wallet.publicKey)) {
+        currentAuthKeypair = wallet;
+      } else if (currentAuthorityPubkey.equals(newAuthority.publicKey)) {
+        currentAuthKeypair = newAuthority;
+      }
+
+      // Unpause deposits if paused
+      if (pdaData.depositPaused && currentAuthKeypair) {
+        await gatewayProgram.methods
+          .setDepositPaused(false)
+          .accounts({
+            pda: pdaAccount,
+            signer: currentAuthorityPubkey,
+          })
+          .signers(currentAuthKeypair === wallet ? [] : [currentAuthKeypair])
+          .rpc();
+      }
+
+      // Test deposit (doesn't require authority - verifies PDA access works)
+      const balanceBefore = await conn.getBalance(pdaAccount);
+      await gatewayProgram.methods
+        .deposit(new anchor.BN(1_000_000_000), Array.from(address), revertOptions)
+        .rpc();
+      const balanceAfter = await conn.getBalance(pdaAccount);
+      expect(balanceAfter - balanceBefore).to.be.gte(1_000_000_000);
+
+      // Verify PDA structure is intact after migration
+      const pda = await gatewayProgram.account.pda.fetch(pdaAccount);
+      expect(pda.bump).to.be.a("number");
+      expect(pda.nominatedAuthority).to.not.be.undefined;
+      expect(pda.nonce).to.not.be.undefined;
+      expect(pda.tssAddress).to.not.be.undefined;
+      expect(pda.authority).to.not.be.undefined;
+      expect(pda.chainId).to.not.be.undefined;
+    });
+
+    it("Should work with SPL token operations after migration", async () => {
```

Comment:
we can do checks with deposit and withdraw after migration to be sure it works

also this migration should be somehow in zetae2e upgrade tests but that is separate PR